VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsAdapter.java

```diff
@@ -80,22 +81,26 @@ public void render(BaseViewHolder tag, DiscussionThread discussionThread) {
         {
             final String text;
             final Iconify.IconValue icon;
+            final boolean iconAtStart;
             @ColorInt
-            final int iconColor;
+            final int indicatorColor;
             if (voteCountsEnabled) {
                 text = Integer.toString(discussionThread.getVoteCount());
                 icon = Iconify.IconValue.fa_plus;
-                iconColor = discussionThread.isVoted() ? edx_brand_primary_base : edx_grayscale_neutral_light;
+                indicatorColor = discussionThread.isVoted() ? edx_brand_primary_base : edx_grayscale_neutral_light;
+                iconAtStart = true;
             } else {
                 text = Integer.toString(discussionThread.getCommentCount());
                 icon = Iconify.IconValue.fa_comment;
-                iconColor = discussionThread.getUnreadCommentCount() == 0 ? edx_grayscale_neutral_light : edx_brand_primary_base;
-
+                indicatorColor = discussionThread.getUnreadCommentCount() == 0 ? edx_grayscale_neutral_light : edx_brand_primary_base;
+                iconAtStart = false;
             }
             holder.discussionPostIndicatorTextView.setText(text);
-            holder.discussionPostIndicatorTextView.setTextColor(iconColor);
-            holder.discussionPostIndicatorIcon.setIcon(icon);
-            holder.discussionPostIndicatorIcon.setIconColor(iconColor);
+            holder.discussionPostIndicatorTextView.setTextColor(indicatorColor);
+
+            final IconDrawable iconDrawable = new IconDrawable(getContext(), icon).sizePx((int) holder.discussionPostIndicatorTextView.getTextSize()).color(indicatorColor);
+            TextViewCompat.setCompoundDrawablesRelativeWithIntrinsicBounds(
+                    holder.discussionPostIndicatorTextView, iconAtStart ? iconDrawable : null, null, iconAtStart ? null : iconDrawable, null);
```

Comment:
This might work better as an `if` `else` statement to avoid having to check the `iconAtStart` state twice. I don't know... it's up to you which you consider most clear.